ortools/sat/csharp/CpSolver.cs

```diff
@@ -28,6 +29,20 @@ namespace Google.OrTools.Sat
  */
 public class CpSolver
 {
+    private static Queue<Term> _cachedQueue;
+
+    [MethodImpl(MethodImplOptions.AggressiveInlining)]
```

Comment:
no need for this. Value() is called after the search. We can document it is not MT-HOT, or make it synchronied.

Reply:
I'll reply here but this is related to other comments as well.

The intent of the queue being static is twofold:
- easy access from multiple classes (CpSolver, Callback, LinearExpression (probably next PR)) (As non static would require met to pass it around)
- share the queue between different solver instances (No need to allocate in the default sequential case)

Making it synchronized works, but will most likely make it slower than the Interlocked.Exchange with the same benefits. (And is only needed if it stays static, if it is not, then there is no point in any protection, since the rest of the code here isn't protected against multithread access)